### PR TITLE
Add send_file from Flask

### PR DESCRIPTION
**Pull request by kx-chen**
Closes https://github.com/pallets/werkzeug/issues/265

### CODE REVIEW

**Comment by kx-chen** [comment]
Should be easier to review now. 03f3fa59ecc95e4cd382fb4286f39e6714d542a9 is the function directly from flask, and 3dfcf0b016788c68d80528e86bc3d6b6e9b510ca shows most of the changes that were made to it.

File: src/werkzeug/wsgi.py [hunk 1]
```diff
@@ -1,19 +1,223 @@
...
 from .urls import uri_to_iri
 from .urls import url_join
 from .urls import url_parse
 from .urls import url_quote
 
 
+def send_file(
+    filename_or_fp,
+    root_path,
+    environ=None,
```
Comment (kx-chen) [inline src/werkzeug/wsgi.py:27]: Not too sure about this approach -- `send_file` in Flask was using the app's `root_path` but we don't have that equivalent in werkzeug?

Comment (davidism) [reply src/werkzeug/wsgi.py:27]: Passing `environ` makes sense. See other review comments about `root_path`.

**Comment by kx-chen** [comment]
Apologies for the lack of progress on this. Will be taking a look and fixing it up more tomorrow.

**Comment by davidism** [comment]
No worries, I'm actually preparing this for merge right now, I don't think there's anything else you need to do.

**Comment by davidism** [comment]
After some more thought, I realized this didn't belong in `wsgi.py`, moved it to `utils.py` for now. Might make a dedicated `response_helpers` module later. Removed the `root_path` code, applied some minor cleanup, reorganized the tests in their own file, added a changelog, and completely rewrote the docs. There's other cleanup I want to do to take advantage of having a new start here, but I'll save that for other PRs.

**Pull request merged**